### DISCUSSION TITLE
Amplify v6 compatibility issue "Error: The package '@aws-amplify/react-native' doesn't seem to be linked."

### DISCUSSION THREAD

**Discussion by truejordan** [expo/expo#25586] [Archived: Questions]
Currently the latest update amplify v6 library is incompatible with expo go.

When amplify is configured in the root layout with:

`
import { Amplify } from 'aws-amplify';
import amplifyconfig from '../src/amplifyconfiguration.json';
Amplify.configure(amplifyconfig);
`
you will have this error:

 ERROR  Error: The package '@aws-amplify/react-native' doesn't seem to be linked. Make sure: 

- You have run 'pod install'
- You rebuilt the app after installing the package
- You are not using Expo Go
, js engine: hermes 

is there a possible work around to get going with the update amplify library while using expo go?

**Comment by ysg00** [comment]
Same issue here:
with the following dependencies in package.json
    "@aws-amplify/react-native": "^1.0.5",
    "@react-native-async-storage/async-storage": "^1.19.8",
    "@react-native-community/netinfo": "^11.1.0",
    "aws-amplify": "^6.0.5",
    "expo": "^49.0.21",
    "expo-status-bar": "~1.6.0",
    "react": "18.2.0",
    "react-native": "0.72.6",
    "react-native-get-random-values": "^1.10.0"
   
Can anyone help please?

**Comment by bum-bum-pav** [comment]
Same error

**Comment by andy-ratsirarson** [comment]
Same error here.

**Comment by truejordan** [comment]
### This will get you going but does not solve issues with using expo go

As of right now expo go is not supported with amplify v6 so you wont be able to use the app on a simulator or device, however you can run your app with commands:

`npx expo run:ios` or `npx expo run:android`

With this you will be able to open your app with the simulator, however for a physical device you will have to plug it in via usb as you would  using regular react native.

You can see set up here:
[https://reactnative.dev/docs/running-on-device?platform=ios](url)

**Reply by andy-ratsirarson** [reply]
This work for now, but it's breaking our flow.
Is there a plan to support this amplify version?

**Reply by truejordan** [reply]
Im unsure but a patch soon would be great

**Reply by kushalpatel2210** [reply]
This worked for me

**Comment by TomasTykac** [comment]
Same error. 
I get same error on simulator and devices as well. Looks like problem come when you are starting integrate API for DB use

**Reply by truejordan** [reply]
have you tired running `npx expo run:ios` or `npx expo run:android` ?

Use either one of those commands instead of `npx expo start`

you can replace npx with with what ever package management you are using i.e "bun, yarn"

**Reply by TomasTykac** [reply]
I tried it with same result. Only way is use older version  (I'm using aws-amplify 5.3.6).

**Reply by nadetastic** [reply]
Hi @TomasTykac I'm from the @aws-amplify team - can you elaborate on your use case? "Integrate API for DB use"

As @truejordan  mentioned, the. latest major version `aws-amplify@6.x.x` no longer supports Expo Go due to the library leveraging native functionality for core Amplify operations that are not available in Expo Go - as it can only use native modules built into the Expo SDK. The recommendation is to run a build using `npx expo run:<android | ios>` which should work without issue

**Reply by TomasTykac** [reply]
Hello @nadetastic  I tried your way with npx expo run:{Android} and  get something different. If I use withAuthenticator from aws-amplify/ui-react I will get a lot of strange errors like "Text string must be rendered within <Text>" which I already have.
But OK with your parameter I'm not getting previous error messages.

**Comment by ddorismond** [comment]
Same error here

**Comment by ddorismond** [comment]
Yes. I get the same result too. Gen 2 is unusable at this point.

**Reply by nadetastic** [reply]
Hi @ddorismond are you running your app with `npx expo run:<android | ios>` instead of `expo start` (expo go)?

**Reply by ddorismond** [reply]
Yes. The error continues to happen at  Amplify.configure(config);.
If I comment it out it works. I tried recreating the Gen2 sandbox instance for good measure, but no luck.

**Comment by alpha-adam** [comment]
Same error, running npx expo run gives this error

duplicate symbol '_OBJC_IVAR_$_JKBigDecimal.bigInteger' in

**Comment by Mxctr1** [comment]
Hey everyone. I don't know what I did but seems that it works for me now. Using Tamagui. I had a project first without tamagui but with Amplify V6 and another one with Tamagui but it seems it did not work. I just copied the package.json of the other files. Not sure what was causing the problem but here is the file : 

`{
  "name": "x",
  "version": "1.0.0",
  "main": "expo-router/entry",
  "scripts": {
    "start": "expo start",
    "android": "expo run:android",
    "ios": "expo run:ios",
    "web": "expo start --web",
    "upgrade:tamagui": "$npm_execpath up '*tamagui*' '@tamagui/*'"
  },
  "dependencies": {
    "@aws-amplify/react-native": "^1.0.5",
    "@react-native-async-storage/async-storage": "1.18.2",
    "@react-native-community/netinfo": "9.3.10",
    "@reduxjs/toolkit": "^1.9.7",
    "@tamagui/animations-react-native": "^1.61.3",
    "@tamagui/babel-plugin": "^1.61.3",
    "@tamagui/config": "^1.61.3",
    "@tamagui/font-inter": "^1.61.3",
    "@tamagui/lucide-icons": "^1.61.3",
    "@tamagui/themes": "^1.61.3",
    "aws-amplify": "^6.0.5",
    "expo": "~49.0.15",
    "expo-constants": "~14.4.2",
    "expo-font": "11.4.0",
    "expo-linear-gradient": "~12.3.0",
    "expo-linking": "~5.0.2",
    "expo-router": "2.0.0",
    "expo-splash-screen": "~0.20.5",
    "expo-status-bar": "~1.6.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-hook-form": "^7.48.2",
    "react-native": "0.72.6",
    "react-native-gesture-handler": "~2.12.0",
    "react-native-get-random-values": "^1.9.0",
    "react-native-reanimated": "~3.3.0",
    "react-native-safe-area-context": "4.6.3",
    "react-native-screens": "~3.22.0",
    "react-native-svg": "13.9.0",
    "react-redux": "^8.1.3",
    "tamagui": "^1.61.3",
    "yarn": "^1.22.21"
  },
  "devDependencies": {
    "@babel/core": "^7.19.3",
    "@google/semantic-release-replace-plugin": "1.1.0",
    "@manypkg/cli": "^0.20.0",
    "@semantic-release/changelog": "6.0.1",
    "@semantic-release/git": "10.0.1",
    "@semantic-release/github": "8.0.6",
    "@semantic-release/npm": "9.0.1",
    "@semantic-release/release-notes-generator": "10.0.3",
    "@types/react": "~18.2.14",
    "@typescript-eslint/eslint-plugin": "^5.49.0",
    "@typescript-eslint/parser": "^5.49.0",
    "babel-plugin-transform-inline-environment-variables": "^0.4.4",
    "conventional-changelog-conventionalcommits": "5.0.0",
    "semantic-release": "19.0.5",
    "typescript": "^5.1.3"
  },
  "overrides": {
    "react-refresh": "~0.14.0"
  },
  "resolutions": {
    "react-refresh": "~0.14.0"
  },
  "private": true
}

`

**Reply by alpha-adam** [reply]
thanks for the post, but it didn't work on my end.

**Comment by TomasTykac** [comment]
I got a solution. 
If you wanna use Amplify 6.0 you have to run app with "npx expo run:Android/ios". You cannot call login form with function "export default withAuthenticate(App)" but you have to use Authenticator hook + provider. (details in amplify docs).

**Reply by Mxctr1** [reply]
That's maybe why mine was working. I am not using the ui default. I build everything by myself and use functions provided by aws

**Reply by mitch-sw** [reply]
@TomasTykac can you link the docs you are referring to? 
I I use the implementation here: https://ui.docs.amplify.aws/react-native/connected-components/authenticator/configuration I am still getting the error.

**Reply by mhraihan** [reply]
after complied it works but I do not know why it does not work with expo go.
also , the error message wasn't helpful too.

**Comment by BergesAronis** [comment]
Is there currently any plan to update expo to resolve the issue? Running into it now as well and wondering when expo will be compatible with aws-amplify again.

**Reply by BergesAronis** [reply]
As a temporary solution can the team provide a known version of `expo` and `aws-amplify` that are compatible? Reverting the dependencies will at least unblock everyone until the newest versions are updated to be compatible.

**Comment by andrewdavidcostello** [comment]
Slightly different error but maybe related?

```
Unable to resolve module @aws-amplify/react-native from /node_modules/@aws-amplify/core/src/utils/amplifyUuid/index.native.ts:
@aws-amplify/react-native could not be found within the project or in these directories: node_modules/@aws-amplify/core/ node_modules node_modules
11// Copyright Amazon.com, Inc. or its affiliates.
All Rights Reserved.
2 | /| SPDX-License-Identifier: Apache-2.0
> 3 | import { loadGetRandomValues } from '@aws-amplify/react-native';
4 | import { v4 } from 'uuid';
5 |
6 | loadGetRandomValues);
```

**Comment by hanna-becker** [comment]
@alpha-adam, I also got duplicate symbol errors with expo run:ios, in the linking phase.
```
❌  duplicate symbol '_OBJC_IVAR_$_JKBigDecimal.bigInteger' in
┌─ libAmplifyRTNCore.a(JKBigDecimal.o)
└─ libRNAWSCognito.a(JKBigDecimal.o)
```
I then removed the amazon-cognito-identity-js package (seeing that the authenticator setup instructions of other JS clients don't list it as a dependency any longer) and that seems to fix it so far. However, the docs of the RN authenticator still say it's needed: https://ui.docs.amplify.aws/react-native/connected-components/authenticator#step-2-install-dependencies

Maybe @nadetastic can comment on whether it's still needed.

I got the above error when just following the RN authenticator setup instructions, installing the latest versions of all the packages listed there.

**Reply by silberistgold** [reply]
I get the same error when building on eas build. Seems to be fixed after removing amazon-cognito-identity-js from dependencies and app seems to work correctly. Is this an error in the setup instructions?

**Comment by darren4flow** [comment]
These packages work for me:

    "@aws-amplify/ui-react-native": "^1.2.27",
    "@react-native-async-storage/async-storage": "1.18.2",
    "aws-amplify": "^5.3.10",
    "aws-sdk": "^2.1458.0",
    "amazon-cognito-identity-js": "^6.3.5",
    "@react-native-community/netinfo": "9.3.10",

**Reply by StephenA1** [reply]
Thank you, much appreciated

**Comment by gpelegrine** [comment]
Using the version: "aws-amplify": "^6.0.11". The error below occurs
![awsAmplifyv6Error](https://github.com/expo/expo/assets/88168923/c654c25c-4d4c-4385-b54c-8c1d8927d43c)

Using version: aws-amplify@"5.3.6". Worked perfectly!

**Comment by sopykt** [comment]
I have same issue but this solve for me. https://docs.expo.dev/develop/development-builds/create-a-build/
By creating the development build using eas and then I simply download and install on my android device. And I run `npx expo start --dev-client --tunnel`. It works like expo go.

**Reply by loalm** [reply]
Thanks, but does this mean I have to make a new build for every small change I make to the app through EAS? In my case I'm using Windows and cannot build iOS locally.

**Reply by danrivett** [reply]
No it doesn't (and I can confirm I did the same thing as @sopykt ).

The video walkthrough that @sopykt linked to is very good and shows how the development build is be used to connect to a local expo server for hot code refreshing as you mention. The video shows this in action.

https://youtu.be/LUFHXsBcW6w

**Reply by sopykt** [reply]
No you don't need to build again everytime you make changes. Just follow the documentations.

**Reply by well-hello-there** [reply]
Can someone confirm that this option also works for iOS devices? Because I tried it without any luck so far. I'm on `"aws-amplify": "^6.0.28"` and `"expo-dev-client": "~3.3.11"`

**Comment by tscherr** [comment]
Any update on this incompatibility? It seems there are two options at the moment: 1) downgrade was-amplify to v5, 2) local builds. Should this be opened as an issue to increase visibility with the expo team?

**Comment by mogusbi** [comment]
It's now nearly the end of February and there's still no fix for this other than the unsatisfactory workaround to downgrade to Amplify 5. Is there any update?

**Comment by dhallX** [comment]
getting same error any updates?

**Comment by SwhiteMHC** [comment]
I am also encountering this error now.

**Comment by JavanPoirier** [comment]
Spent a few hours digging into this and was able to resolve it. First off, I am in a monorepo that may be adding some complexities as I am using amplify in a component package `app` which exports screens that are imported and directly exported as pages using Next,js and Expo's file based routing.

I suspect the main culprit was that I had `@aws-amplify/react-native` installed in my `app` package. This was most likely installed using yarn (yarn install @aws-amplify/react-native). This is incorrect. The linking process of which this error directly refers to is performed by the expo cli. Expo packages should be installed using the cli `(npx|yarn) expo install`. By ensuring `@aws-amplify/react-native` was installed in my expo app directory and not my component package using the expo cli `expo install @aws-amplify/react-native` I was able to get this working. 

Also, I have this working with Expo Go and do not require a compiled development build for this to work like some suggest here.

**Reply by raphaeljoer** [reply]
I did but now I'm getting this error:

`
iOS Bundling failed 34509ms
Unable to resolve "@aws-amplify/react-native" from "node_modules/@aws-amplify/core/src/utils/amplifyUuid/index.native.ts"
`

`"aws-amplify": "^6.0.28",`

**Comment by bracki** [comment]
@JavanPoirier do you have a public repo where we could see how you solved the problem?

**Comment by jewells07** [comment]
I'm running into same issue.
```javascript
 ERROR  [Unknown: An unknown error has occurred.]
 ERROR  [Error: The package '@aws-amplify/react-native' doesn't seem to be linked. Make sure:
 ```

**Comment by francescovenica** [comment]
I think aws dropped expo compatibility, so there isn't a lot we can do

https://docs.amplify.aws/gen1/react-native/build-a-backend/troubleshooting/migrate-from-javascript-v5-to-v6/#dropping-support-for-expo-go

**Reply by ysg00** [reply]
Is this just for gen 1?

**Reply by ysg00** [reply]
For gen 2 it said
<img width="1256" alt="image" src="https://github.com/user-attachments/assets/9f7fe0c4-39b9-4a5a-8127-e4c4b65b675f">

**Reply by stewartmoreland** [reply]
This is only for Expo Go. To use Amplify's React Native components, you'll need to create a dev build using `npx expo prebuild` & `npx expo run:ios`. If you want to test on physical devices, run a development build through EAS. This will still connect to Metro for live component rendering but will include the native modules used in the `@aws-amplify/ui-react-native` & `@aws-amplify/react-native` libraries.
